scripts/sanitycheck

```diff
@@ -2675,6 +2677,8 @@ def parse_arguments():
         which will ultimately disable ccache.
         """
     )
+    parser.add_argument("--gcov-tool", default="gcov",
```

Comment:
since this script is specific to zephyr, can we take the default gcov binary from the SDK?.
We don't need to worry about host tools.

Reply:
no we can't, the Zephyr SDK is not the only way we build Zephyr, we also need to support the fact the native_posix is built using the host toolchain, and not the SDK.
Sanitycheck itself also does not know anything about the toolchain, To make all of this work smoothly, the gcov data needs to be generated by the application cmakefile which then would used whatever comes with the toolchain, regardless if it is the SDK or the host toolchain, but this is a story for another time.

Reply:
oh k. my only concern was that of the shippable.